core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1374,12 +1374,15 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * This can result in canceling pending requests or filing additional requests.
    * This is currently only supported in YARN mode. Return whether the request is received.
    */
-  private[spark] override def requestTotalExecutors(numExecutors: Int): Boolean = {
+  private[spark] override def requestTotalExecutors(
+      numExecutors: Int,
+      localityAwarePendingTasks: Int,
+      preferredLocalityToCount: scala.Predef.Map[String, Int]): Boolean = {
```

Comment:
Hi @squito , thanks a lot for your explanation, since the previous code already import `scala.collection.Map`, so here I have to write a full qualified name here, I will change to immutable map :).